OptimizelySDK/Optimizely.cs

```diff
@@ -308,5 +296,226 @@ public Variation GetForcedVariation(string experimentKey, string userId)
+        /// Gets feature variable value for given type.
+        /// </summary>
+        /// <param name="featureKey">The feature flag key</param>
+        /// <param name="variableKey">The variable key</param>
+        /// <param name="userId">The user ID</param>
+        /// <param name="userAttributes">The user's attributes.</param>
+        /// <param name="variableType">Variable type</param>
+        /// <returns>string | Feature variable value</returns>
+        public virtual string GetFeatureVariableValueForType(string featureKey, string variableKey, string userId, 
+            UserAttributes userAttributes, FeatureVariable.VariableType variableType)
+        {
+            if (string.IsNullOrEmpty(featureKey))
+            {
+                Logger.Log(LogLevel.ERROR, "Feature flag key must not be empty.");
+                return null;
+            }
+
+            if (string.IsNullOrEmpty(variableKey))
+            {
+                Logger.Log(LogLevel.ERROR, "Variable key must not be empty.");
+                return null;
+            }
+
+            if (string.IsNullOrEmpty(userId))
+            {
+                Logger.Log(LogLevel.ERROR, "User ID must not be empty.");
+                return null;
+            }
+
+            var featureFlag = Config.GetFeatureFlagFromKey(featureKey);
+            if (string.IsNullOrEmpty(featureFlag.Key))
+                return null;
+
+            var featureVariable = featureFlag.GetFeatureVariableFromKey(variableKey);
+
+            if (featureVariable == null)
+            {
+                Logger.Log(LogLevel.ERROR,
+                    $@"No feature variable was found for key ""{variableKey}"" in feature flag ""{featureKey}"".");
+                return null;
+            }
+            else if (featureVariable.Type != variableType)
+            {
+                Logger.Log(LogLevel.ERROR,
+                    $@"Variable is of type ""{featureVariable.Type}"", but you requested it as type ""{variableType}"".");
+                return null;
+            }
+
+            var variableValue = featureVariable.DefaultValue;
+            var variation = DecisionService.GetVariationForFeature(featureFlag, userId, userAttributes);
+
+            if (variation != null && variation.GetFeatureVariableUsageFromId(featureVariable.Id) != null)
+            {
+                var featureVariableUsageInstance = variation.GetFeatureVariableUsageFromId(featureVariable.Id);
+                
+                variableValue = featureVariableUsageInstance.Value;
+                Logger.Log(LogLevel.INFO,
+                    $@"Returning variable value ""{variableValue}"" for variation ""{variation.Key}"" of feature flag ""{featureFlag.Key}"".");
+            }
+            else
+            {
+                Logger.Log(LogLevel.INFO,
+                    $@"User ""{userId}"" is not used in any variation for feature flag ""{featureFlag.Key}"", returning default value ""{variableValue}"".");
```

Comment:
📝 Remove 'used'. 
📝 This message is misleading, This will be logged even when a variation does exist, but no variable usage is found for the variation. See PHP for another log message :
"Variable '{$variableKey}' is not used in variation '{$variation->getKey()}' ". 
                    "returning default value '{$variable_value}'.");

Reply:
@oakbani Do we need to log specific variation on which the variable usage does not get found? Also, can we verify it in unit tests? I think it's better to avoid logging the specific variation id as the end result would be same in both cases. i.e. default value.

Let's discuss it over Skype.

Reply:
Done